File: project/research/general_n/2026-09-17-d5-defect11-closure-v1/check_d11_singletons.py
from pathlib import Path
import itertools, json, hashlib
verts=range(5)
edges=list(itertools.combinations(verts,2))
def threat(mask,S):
    S=set(S); adj=[set() for _ in verts]; Q=[]
    for b,(i,j) in enumerate(edges):
        if mask>>b&1:
            adj[i].add(j); adj[j].add(i); Q.append((i,j))
    out=set()
    for i,j in Q:
        for t,s in ((i,j),(j,i)):
            if t not in S and s in S and (adj[t]&S)=={s}:
                out.add((min(i,j),max(i,j))); break
    return out
def solve(mask):
    Q=[e for b,e in enumerate(edges) if mask>>b&1]
    if not Q: return (0,0)
    idx={e:i for i,e in enumerate(Q)}; full=(1<<len(Q))-1
    fam=[]
    for r in range(1,6):
        c=0 if r==5 else 4-r; sing=1 if r==1 else 0
        for S in itertools.combinations(verts,r):
            bm=0
            for e in threat(mask,S): bm |= 1<<idx[e]
            if bm: fam.append((bm,c,sing))
    INF=(99,99); dp=[INF]*(1<<len(Q)); dp[0]=(0,0)
    for st in range(1<<len(Q)):
        if dp[st]==INF: continue
        for bm,c,sing in fam:
            ns=st|bm; cand=(dp[st][0]+c,dp[st][1]+sing)
            if cand<dp[ns]: dp[ns]=cand
    return dp[full]
rows=[]; types={}
for mask in range(1<<10):
    tau,mins=solve(mask); mu=10-mask.bit_count(); total=2*mu+tau
    deg=[0]*5
    for b,(i,j) in enumerate(edges):
        if mask>>b&1: deg[i]+=1; deg[j]+=1
    row=(mask,mu,tau,mins,total,*sorted(deg)); rows.append(row)
    if total==11:
        k=str((mu,tau,mins,tuple(sorted(deg))))
        types[k]=types.get(k,0)+1
stream=''.join(' '.join(map(str,r))+'\n' for r in rows).encode()
summary={'status':'PASS_D11_OPTIMAL_COVER_SINGLETON_CLASSIFICATION','tight_graphs':1024,'d11_graphs':sum(r[4]==11 for r in rows),'minimum_types':types,'all_d11_require_singleton':all(r[3]>=1 for r in rows if r[4]==11),'row_sha256':hashlib.sha256(stream).hexdigest()}
Path(__file__).with_name('PYTHON_ROWS.txt').write_bytes(stream)
Path(__file__).with_name('CHECK_SUMMARY.json').write_text(json.dumps(summary,indent=2,sort_keys=True)+'\n')
print(json.dumps(summary,indent=2,sort_keys=True))

File: project/research/general_n/2026-09-17-d5-defect11-closure-v1/check_d11_singletons.cpp
#include <bits/stdc++.h>
using namespace std;
int main(){
 vector<pair<int,int>> E;for(int i=0;i<5;i++)for(int j=i+1;j<5;j++)E.push_back({i,j});
 map<string,int> types; int d11=0; bool allsing=true;
 for(int mask=0;mask<(1<<10);++mask){
  vector<pair<int,int>>Q;int deg[5]={};
  for(int b=0;b<10;b++)if(mask>>b&1){Q.push_back(E[b]);deg[E[b].first]++;deg[E[b].second]++;}
  int m=Q.size(),full=(1<<m)-1; struct F{int bm,c,s;};vector<F>fam;
  for(int S=1;S<32;S++){
   int r=__builtin_popcount((unsigned)S),c=(r==5?0:4-r),sing=(r==1),bm=0;
   for(int qi=0;qi<m;qi++){
    auto [i,j]=Q[qi];bool th=false;
    for(auto pr:{pair<int,int>{i,j},pair<int,int>{j,i}}){
     int t=pr.first,s=pr.second;if((S>>t&1)||!(S>>s&1))continue;
     int cnt=0;
     for(int v=0;v<5;v++)if(S>>v&1){
      int a=min(t,v),b=max(t,v),bi=-1;
      for(int k=0;k<10;k++)if(E[k]==make_pair(a,b)){bi=k;break;}
      if(bi>=0&&(mask>>bi&1))cnt++;
     }
     if(cnt==1){th=true;break;}
    }
    if(th)bm|=1<<qi;
   }
   if(bm)fam.push_back({bm,c,sing});
  }
  pair<int,int> INF={99,99};vector<pair<int,int>>dp(1<<m,INF);dp[0]={0,0};
  for(int st=0;st<(1<<m);st++)if(dp[st]!=INF)for(auto f:fam){
   int ns=st|f.bm;pair<int,int>cand={dp[st].first+f.c,dp[st].second+f.s};
   if(cand<dp[ns])dp[ns]=cand;
  }
  int tau=dp[full].first,mins=dp[full].second,mu=10-m,total=2*mu+tau;
  sort(deg,deg+5);
  if(total==11){d11++;allsing=allsing&&(mins>=1);string k=to_string(mu)+":"+to_string(tau)+":"+to_string(mins)+":";for(int x:deg){k+=to_string(x);}types[k]++;}
  cout<<mask<<' '<<mu<<' '<<tau<<' '<<mins<<' '<<total;for(int x:deg)cout<<' '<<x;cout<<'\n';
 }
 cerr<<"d11="<<d11<<" all_require_singleton="<<allsing<<"\n";for(auto &[k,v]:types)cerr<<k<<" "<<v<<"\n";
}
